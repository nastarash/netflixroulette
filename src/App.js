import React, { Fragment } from 'react';
import Header from './components/header/Header';
import Footer from './components/footer/Footer';
import SearchButton from './components/searchButton/SearchButton'

function App() {
  return (
    <Fragment>
      <Header />
      <SearchButton />
      <Footer />
    </Fragment>
  );
}

export default App;
