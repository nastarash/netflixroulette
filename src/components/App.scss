@import '~CommonStyles';

html,
body {
  margin: 0;
  padding: 0;
  height: 100%;


}

.page-container {
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  background-color: $main-bg-color;

  &__content {
    padding: 30px 50px 50px 50px;
    flex-grow: 1;
    background-color: $light-bg-color;
  }
}
