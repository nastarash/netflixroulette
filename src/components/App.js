import React, { Component } from 'react';
import { Provider } from 'react-redux';
import './App.scss';
import { hot } from 'react-hot-loader';

import { AppErrorBoundary } from 'CommonComponents/AppErrorBoundary';
import { PageHeader } from 'CommonComponents/PageHeader';
import { PageFooter } from 'CommonComponents/PageFooter';
import { FilmResultsContainer } from 'CommonComponents/FilmResultsContainer';

import { Route, Switch } from 'react-router-dom';
import { SingleFilmContainer } from 'CommonComponents/SingleFilmContainer';
import { NotFound } from './NotFound';

export class App extends Component {
  render() {
    const { Router, location, context, store } = this.props;

    return (
      <AppErrorBoundary>
        <Provider store={store}>
          <Router location={location} context={context}>
            <div className="page-container">
              <Route
                render={props => (
                  <PageHeader {...props}>
                    <Route
                      exact
                      path="/film/:id?"
                      component={SingleFilmContainer}
                    />
                  </PageHeader>
                )}
              />
              <div className="page-container__content">
                <Switch>
                  <Route
                    exact
                    path={[
                      '/search/:searchString',
                      '/search/',
                      '/search',
                      '/',
                      '/film/:id?',
                      ,
                    ]}
                    component={FilmResultsContainer}
                  />
                  <Route path="/404" component={NotFound} />
                  <Route component={NotFound} />
                </Switch>
              </div>
              <PageFooter />
            </div>
          </Router>
        </Provider>
      </AppErrorBoundary>
    );
  }
}

export default hot(module)(App);
