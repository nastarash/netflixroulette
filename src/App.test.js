import React from 'react';
import App from './components';
import renderer from 'react-test-renderer';

it("render correctly", () => {
    const AppComponent = renderer.create(<App />).toJSON();
    expect(AppComponent).toMatchSnapshot();
});