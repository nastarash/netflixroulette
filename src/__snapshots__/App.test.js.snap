// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
Array [
  <h1
    id="helloWorld"
  >
    Hello World! Task 1 within React Mentoring Programm
  </h1>,
  <li
    className="createElement"
  >
    Component created by React.createElement
  </li>,
  <li>
    Component created by React.Component
  </li>,
  <li>
    Component created by React.PureComponent
  </li>,
  <li>
    Component created by using function
  </li>,
]
`;
